concourse/steps/component_descriptor_util.py

```diff
@@ -37,3 +43,79 @@ def parse_component_descriptor(
       return component_descriptor
     else:
         raise NotImplementedError(schema_version)
+
+
+def component_descriptor_from_component_descriptor_path(cd_path: str) -> cm.ComponentDescriptor:
+
+    if not os.path.isfile(cd_path):
+        raise FileNotFoundError(
+            f'{os.path.abspath(cd_path)=} not found'
+        )
+
+    descriptor_v2 = cm.ComponentDescriptor.from_dict(
+        ci.util.parse_yaml_file(cd_path)
+    )
+    logger.info(f'found component-descriptor (v2) at {cd_path=}')
+    return descriptor_v2
+
+
+def component_descriptor_from_ctf_path(ctf_path: str) -> cm.ComponentDescriptor:
+
+    if not os.path.exists(ctf_path):
+        raise FileNotFoundError(
+            f'{os.path.abspath(ctf_path)=} not found'
+        )
+
+    if not os.path.isfile(ctf_path):
+        raise ValueError(
+            f'{os.path.abspath(ctf_path)=} is not a file'
+        )
+
+    component_descriptors = [
+        cd
+        for cd in cnudie.util.component_descriptors_from_ctf_archive(ctf_path)
+    ]
+
+    if len(component_descriptors) == 0:
+        raise RuntimeError(
+            f'No component descriptor found in ctf archive at {os.path.abspath(ctf_path)}'
+        )
+    if len(component_descriptors) > 1:
+        raise RuntimeError(
+            f'More than one component_descriptor found at {os.path.abspath(ctf_path)}'
+        )
+    logger.info(f'found component-descriptor (v2) in {ctf_path=}')
+    return component_descriptors[0]
+
+
+def component_descriptor_from_dir_path(dir_path: str) -> cm.ComponentDescriptor:
+
+    if not os.path.isdir(dir_path):
+        raise NotADirectoryError(
+            f'{os.path.abspath(dir_path)=} is no directory'
+        )
+
+    v2_outfile = os.path.join(
+        dir_path,
+        component_descriptor_fname(
+            schema_version=cm.SchemaVersion.V2,
+        ),
+    )
+    ctf_out_path = os.path.abspath(
+        os.path.join(
+            dir_path,
+            product.v2.CTF_OUT_DIR_NAME,
+        )
+    )
+
+    have_ctf = os.path.exists(ctf_out_path)
+    have_cd = os.path.exists(v2_outfile)
+
+    if not have_ctf ^ have_cd:
+        logger.error(f'exactly one of {ctf_out_path=}, {v2_outfile=} must exist, not both.')
```

Comment:
omit `not both`
1stly, it is already clear from "exactly one of", 2ndly, the error will also be risen if both are falsy